helm-charts/splunk-otel-collector/templates/daemonset.yaml

```diff
@@ -251,7 +251,11 @@ spec:
         imagePullPolicy: {{ .Values.image.otelcol.pullPolicy }}
         {{- if or $agent.securityContext (and (eq (include "splunk-otel-collector.logsEnabled" $) "true") (eq .Values.logsEngine "otel")) }}
         securityContext:
-          {{- include "splunk-otel-collector.securityContext" (dict "isWindows" .Values.isWindows "securityContext" $agent.securityContext "setRunAsUser" true) | nindent 10 }}
+          {{- if .Values.isWindows}}
+          {{- include "splunk-otel-collector.windowsSecurityContext" (dict "securityContext" $agent.securityContext "setRunAsUser" true) | nindent 10 }}
+          {{- else }}
+          {{- include "splunk-otel-collector.securityContext" (dict "securityContext" $agent.securityContext "setRunAsUser" true) | nindent 10 }}
+          {{- end}}
```

Comment:
Can we incorporate this login in the `splunk-otel-collector.securityContext` helper instead of having two of them?

Reply:
Hmm, yes, I didn't go for it because I thought it would be quite messy, but let me give it a shot.

Reply:
Done, please review